### PR TITLE
Signature refactoring

### PR DESCRIPTION
**Pull request by reka**
Removed boolean parameter validateSignature from the function ResponseMapper.map
Introduced a new function mapInclSignature instead.
Advantages:
 - No boolean parameter, but a "decorator" function calling a base function instead.
 - It became more obvious that the response->isValidSignature returns true, if no signature validation happens
 - This change automatically fixed the unit tests where a transaction object was regarded as a boolean parameter.

Problem with the current code:
 - TransactionServiceUTest: code duplication between the methods mockProcessingRequest and mockProcessingRequestInclSignature

### CODE REVIEW

File: src/Mapper/ResponseMapper.php [hunk 1]
```diff
@@ -78,25 +83,39 @@ public function __construct(Config $config)
     }
 
     /**
+     * @param string $response
+     * @return Response
+     * @throws \Wirecard\PaymentSdk\Exception\MalformedResponseException
```
Comment (horfic) [inline src/Mapper/ResponseMapper.php]: return after throws

File: src/Mapper/ResponseMapper.php [hunk 2]
```diff
@@ -78,25 +83,39 @@ public function __construct(Config $config)
...
      */
-    public function map($response, $validateSignature = false, Transaction $transaction = null)
+    public function map($response, Transaction $transaction = null)
     {
         $this->transaction = $transaction;
 
         // If the response is encoded, we need to first decode it.
         $decodedResponse = base64_decode($response);
-        $response = (base64_encode($decodedResponse) === $response) ? $decodedResponse : $response;
+        $this->encodedResponse = (base64_encode($decodedResponse) === $response) ? $decodedResponse : $response;
```
Comment (horfic) [inline src/Mapper/ResponseMapper.php]: why encodedResponse? its a string containing xml

File: src/TransactionService.php [hunk 3]
```diff
@@ -160,6 +161,7 @@ public function handleNotification($xmlResponse)
      * @throws UnconfiguredPaymentMethodException
      * @throws \RuntimeException
      * @return FailureResponse|InteractionResponse|SuccessResponse|Response
+     * @throws \InvalidArgumentException
```
Comment (horfic) [inline src/TransactionService.php]: return after throws

File: src/Mapper/ResponseMapper.php (hunk 2 above)
Comment (reka) [inline src/Mapper/ResponseMapper.php]: Yup, "encodedResponse" is a bad name.
Corrected to "xmlResponse".

**Pull request merged**